supervised/automl.py

```diff
@@ -866,74 +461,124 @@ def load_progress(self):
+            )
 
-            self.load_progress()
+        # Enforce column names
+        # Enforce X_train columns to be string
+        X.columns = X.columns.astype(str)
 
-            if self._fit_level == "finished":
-                print("AutoML is trained. Skipping fit step ...")
-                return
+        X.reset_index(drop=True, inplace=True)
+
+        if y is None:
+            return X
 
-            # if self._best_model is not None:
-            #    print("Best model is already set, no need to run fit. Skipping ...")
-            #    return
+        # Check if y is np.ndarray, transform to pd.Series
+        if isinstance(y, np.ndarray):
+            y = check_array(y, ensure_2d=False)
+            y = pd.Series(np.array(y), name="target")
+        # if pd.DataFrame, slice first column
+        elif isinstance(y, pd.DataFrame):
+            y = np.array(y.iloc[:, 0])
+            y = check_array(y, ensure_2d=False)
+            y = pd.Series(np.array(y), name="target")
+
+        X, y = ExcludeRowsMissingTarget.transform(X, y, warn=True)
+
+        return X, y
+
+    def _fit(self, X, y):
+        """Fits the AutoML model with data"""
+        # Validate input and build dataframes
+        X, y = self._build_dataframe(X, y)
+
+        self.n_features = X.shape[1]
+        self.n_classes = len(np.unique(y[~pd.isnull(y)]))
+
+        # Get attributes (__init__ params)
+        self._mode = self._get_mode()
+        self._ml_task = self._get_ml_task()
+        self._tuning_mode = self._get_tuning_mode()
+        self._results_path = self._get_results_path()
+        self._total_time_limit = self._get_total_time_limit()
+        self._model_time_limit = self._get_model_time_limit()
+        self._algorithms = self._get_algorithms()
+        self._train_ensemble = self._get_train_ensemble()
+        self._stack_models = self._get_stack_models()
+        self._eval_metric = self._get_eval_metric()
+        self._validation_strategy = self._get_validation_strategy()
+        self._verbose = self._get_verbose()
+        self._explain_level = self._get_explain_level()
+        self._golden_features = self._get_golden_features()
+        self._feature_selection = self._get_feature_selection()
+        self._start_random_models = self._get_start_random_models()
+        self._hill_climbing_steps = self._get_hill_climbing_steps()
+        self._top_models_to_improve = self._get_top_models_to_improve()
+        self._random_state = self._get_random_state()
+
+        # Must pass copies otherwise, X and y get lost
+        self._save_data(deepcopy(X), deepcopy(y))
+
+        self.verbose_print(f"AutoML current directory: {self._results_path}")
+        self.verbose_print(f"AutoML current task: {self._ml_task}")
+        self.verbose_print(f"AutoML will use algorithms : {self._algorithms}")
+        self.verbose_print(f"AutoML will optimize for metric : {self._eval_metric}")
+
+        try:
+            self.load_progress()
```

Comment:
There is a missing check if `AutoML` was already fitted.

Reply:
Part of the scikit-learn ideology is that a model can be fitted `n` times. Any `fit()` call should be independent from the previous calls.

Reply:
But in the case of AutoML the model training can be many hours. The training can be interrupted and then restored. In this case, I won't keep the scikit-learn ideology.

Reply:
Right now `fit()` can be called multiple times, and it will continue previous training.

Reply:
Fixed this. For now `fit` can only be called once